src/main/java/appeng/fluids/parts/PartFluidStorageBus.java

```diff
@@ -537,4 +541,26 @@ public ItemStack getItemStackRepresentation() {
     public GuiBridge getGuiBridge() {
         return GuiBridge.GUI_STORAGEBUS_FLUID;
     }
+
+    // TODO: 1/28/2024 Unify both methods.
+    /**
+     * Filters the changes to only include items that pass the storage filter.
+     * Optimally, this should be handled by the underlying monitor.
+     *
+     * @see appeng.parts.misc.PartStorageBus#filterChanges
+     */
+    protected Iterable<IAEFluidStack> filterChanges(Iterable<IAEFluidStack> change) {
+        var storageFilter = this.getConfigManager().getSetting(Settings.STORAGE_FILTER);
+        if (storageFilter == StorageFilter.EXTRACTABLE_ONLY) {
+            var filteredList = new ArrayList<IAEFluidStack>();
```

Comment:
Since we don't get to know what we're passed by design, only that it's an iterable, I assume that a read-only collection can be passed just fine.

I could replace this with the iterator#remove pattern, but that would be risky.